### PR TITLE
Refactor pages/give/review

### PR DESCRIPTION
**Pull request by delianides**
SYS-2513

### CODE REVIEW

**Comment by jbaxleyiii** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://315487f9-0ace-4811-9d04-682750b93304.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/NewSpring/Holtzman/branch/refactor-give-review-2513/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

File: imports/pages/give/review/Layout.js [hunk 1]
```diff
@@ -72,11 +76,15 @@ export default class Layout extends Component {
     return text;
   }
 
-  icon = (icon) => (
+  icon = (icon: string): any => (
     <AccountType width="30px" height="21px" type={icon} />
   )
 
   render() {
+    if (!this.props.data) {
```
Comment (jbaxleyiii) [inline imports/pages/give/review/Layout.js]: lets one line this

File: imports/pages/give/review/__tests__/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "campus": "Web",
+      "campusId": 20,
+      "email": "web@newspring.cc",
+      "firstName": "Jeff",
+      "lastName": "Goldblumm",
+    }
+  } />
+`;
+
+>>>>>>> 48d0c7b... added give/review tests and snapshots
```
Comment (jbaxleyiii) [inline imports/pages/give/review/__tests__/__snapshots__/index.js.snap]: is this a git merge conflict?

File: imports/pages/give/review/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import { shallowToJson } from "enzyme-to-json";
+import { reset, startBuffering } from "aphrodite/lib/inject";
+
+import { Template } from "../";
+
+export const sampleData = {
+  "data": {
+    "payment": {
+      "icon": "Visa",
+      "last4": "1112",
```
Comment (jbaxleyiii) [inline imports/pages/give/review/__tests__/index.js]: @delianides in master, support for this is removed (just the last4) so your snapshots may need adjusting 👍

File: imports/pages/give/review/index.js [hunk 4]
```diff
@@ -8,21 +10,26 @@ import Success from "../../../blocks/give/Success";
...
+  props: ITemplate;
 
   componentWillMount() {
     const giveData = this.getGiveData();
+
+    if (!giveData) return;
+
     const { dispatch } = this.props;
-    // hide nav
+  //   // // hide nav
```
Comment (jbaxleyiii) [inline imports/pages/give/review/index.js]: double comments?

File: imports/pages/give/review/index.js [hunk 5]
```diff
@@ -8,21 +10,26 @@ import Success from "../../../blocks/give/Success";
...
     dispatch(navActions.hide());
-    // store payment personal and billing
+    // // store payment personal and billing
```
Comment (jbaxleyiii) [inline imports/pages/give/review/index.js]: double comments?

File: imports/pages/give/review/index.js [hunk 6]
```diff
@@ -35,7 +42,7 @@ class Template extends Component {
     dispatch(giveActions.setUserId(giveData.userId));
   }
 
-  onSubmit = (event) => {
+  onSubmit = (event: Function) => {
```
Comment (jbaxleyiii) [inline imports/pages/give/review/index.js]: event should be an `Event` type

File: imports/pages/give/review/index.js [hunk 7]
```diff
@@ -65,27 +72,28 @@ class Template extends Component {
...
 
-    strVal = strVal.replace(/\B(?=(\d{3})+(?!\d))/g, ",");
-    return `$${strVal}`;
+    amount = amount.replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+    return `$${amount}`;
   }
 
   render() {
-    const giveData = this.getGiveData();
+    // const giveData = this.getGiveData();
```
Comment (jbaxleyiii) [inline imports/pages/give/review/index.js]: if we aren't using it, let's remove this line

File: imports/pages/give/review/index.js (hunk 5 above)
Comment (delianides) [inline imports/pages/give/review/index.js]: double comments all the way! 🌈

**Comment by jbaxleyiii** [comment]
@delianides do you feel good about this?

**Pull request merged**